### DISCUSSION TITLE
/posts/how-to-setup-aerospace-tiling-window-manager

### DISCUSSION THREAD

**Discussion by giscus** [josean-dev/sveltekit-blog#17] [General]
# /posts/how-to-setup-aerospace-tiling-window-manager

Use this guide along with my youtube video to setup and use the aerospace tiling window manager on your Mac. Aerospace has been the best tiling window manager I've tried on macOS after struggling with Yabai & Amethyst.

https://www.josean.com/posts/how-to-setup-aerospace-tiling-window-manager

**Comment by lpc84** [comment]
Thanks, I really enjoy your videos

**Comment by hareesh2581** [comment]
Hello Brother,
Love your content on Youtube and Blogs, can you please make a video on how you setup your website with Hugo also share your template, talk about how you setup your website?


I am not a software Engineer, if you think it is easy to just download your theme, modify files and start created markdown files to publish blogs, it will be really helpful as i feel wordpress is veryslow.

**Reply by patterninterrupt** [reply]
Not sure if you're looking to get into software development or not. If not, probably not the move to manage your own site that way. HUGO isn't super complex or anything, but creating your site and then having to manage deployments, where it's hosted, dns, and then any customization that you want to do is a lot still for non-devs.

If you have an interesting in learning, I think it's a great idea.

**Comment by lukeurban** [comment]
I've been missing i3 on Mac. Thanks to your video, I've installed it immediately. I have to say dev made excellent job with the software. I thought it would be hard to adjust, but it took me like 10 mins. Amethyst was my manager of choice up to this point, but now I have a new champion!
Thank you!

**Comment by Shivansh-25** [comment]
Thank you so much for all the content on youtube and also these blogs they are very helpful to go back to things if I ever forget them 
Great content as usual and thanks for the neovim setup man ....

**Comment by martinengelke** [comment]
Open a new finder tab will cause the tiling to acting weird.

**Comment by uchoa** [comment]
Have you ever figured out how to mimic i3's $mod+enter keybinding to create a new WezTerm window in AeroSpace?

**Reply by uchoa** [reply]
I'm aware of the AppleScript example documented in the site, but couldn't yet figure out what to "tell" AeroSpace in order to always have a new window when pressing alt+enter.

**Reply by uchoa** [reply]
Found it:
`alt-enter = 'exec-and-forget /usr/local/bin/wezterm cli spawn --new-window'`

**Reply by uchoa** [reply]
Even better option:
`cmd-enter = 'exec-and-forget open -n /Applications/WezTerm.app'`

**Comment by liuxiriver** [comment]
Thanks Man. really nice doc and video, enjoy a lot!

**Comment by amine2233** [comment]
nice video,

but aerospace is no sense software, not intuitive, need to keep in mind where app is located because doesn't work with mac os workspace, sorry but it's not good windows manager,

**Reply by martinengelke** [reply]
which one do you recommend?

**Reply by amine2233** [reply]
i recommend, yabai without disabling system integrity protection, you can use it easily with native macOS workspace and using multiple monitor. 

you don't have a huge benefice when you disabling system integrity protection.

**Comment by martinengelke** [comment]
yabai says:

The following features of yabai require System Integrity Protection to be (partially) disabled:

focus/move/swap/create/destroy space
remove window shadows
enable window transparency
enable window animations
scratchpad windows
control window layers (make windows appear topmost or on the desktop)
sticky windows (make windows appear on all spaces on the display that contains the window)
toggle picture-in-picture for any given window


quite a lot?

**Comment by slocoro** [comment]
Thanks for creating this guide. I'm using a UK keyboard which has the # key mapped to alt-3 making it impossible to use with Aerospace as trying to type # will switch workspaces. Anyone faced the same issue? I've remapped switching workspaces to alt+ctrl+<number> but wondering if people found a better option.

**Comment by dberna2** [comment]
Hi. I love your tutorials ❤️

One question aeroSpace is compatible with sketchybar?

I would like have the style of sketchybar but with the benefits of aeroSpace to change between workspaces

**Comment by Palanikannan1437** [comment]
This is beautiful stuff! Absolutely love aerospace, hated the macos delays while switching spaces (even with reduced motion), this was amazing, moving from yabai to aerospace now!
